addons/udes_stock_cron/models/stock_move_line.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, _
+from collections import defaultdict
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def any_destination_locations_default(self):
+        """Checks if all location_dest_id's are the picks default
+           location_dest_id of the picking.
+        """
+        default_dest = self.mapped("picking_id.location_dest_id")
+        default_dest.ensure_one()
+        return any(ml.location_dest_id == default_dest for ml in self)
+    
+    def _get_all_products_quantities(self):
+        """This function computes the different product quantities for the given move_lines
+        """
+        res = defaultdict(int)
+        for move_line in self:
+            res[move_line.product_id] += move_line.product_uom_qty
```

Comment:
If this is called on two different pickings, one with UoM of things in 10's the other in things in 1, and they have prouct_uom_qty = 10 each.
This may not be as useful, because if would say {apple: 20} but in terms of the actual quantity of apples  (if they are in units) it is  110. 
Sowe could:
a. Use product_qty so then it is in the UoM of the product (would give 110 here)
b. raise error if ml with the same products have differnent UoMs
c. Add to the doc string

Not sure how it is used, haven't checked but just something to consider.